### PR TITLE
callback codemirror ext setup

### PR DESCRIPTION
**Pull request by rouzwelt**

### CODE REVIEW

File: crates/common/src/lsp_services.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            },
+        };
+
+        parse_rainlang_fork(frontmatter, &rainlang, rpc_url, block_number)
+            .await
+            .map_or_else(
+                |e| {
+                    vec![Problem {
+                        msg: e.to_string(),
+                        position: [0, 0],
```
Comment (mattyg) [inline crates/common/src/lsp_services.rs:73]: same note as above :)

File: tauri-app/src/lib/utils/langServices.ts [hunk 2]
```diff
@@ -52,8 +52,38 @@ export async function parseDotrain(dotrain: RainDocument): Promise<Problem[]> {
...
+ * Provides problems callback by invoking related tauri command
+ */
+export async function problemsCallback(textDocument: TextDocumentItem): Promise<Problem[]> {
+  try {
+    return await invoke('provide_problems', { textDocument, rpcUrl: get(rpcUrl), blockNumber: get(forkBlockNumber) });
+  }
+  catch (err) {
+    return [{
+      msg: typeof err === "string" ? err : err instanceof Error ? err.message : "something went wrong!",
+      position: [0, 0],
```
Comment (mattyg) [inline tauri-app/src/lib/utils/langServices.ts:71]: same note as above :)

**Review by mattyg — CHANGES_REQUESTED** [review]
just requesting changes to default error positions

File: crates/common/src/lsp_services.rs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    if !all_problems.is_empty() {
+        all_problems.iter().map(|&v| v.clone()).collect()
+    } else {
+        let rainlang = match rain_document.compose(&ORDERBOOK_ORDER_ENTRYPOINTS) {
+            Ok(v) => v,
+            Err(e) => match e {
+                ComposeError::Reject(msg) => {
+                    return vec![Problem {
+                        msg,
+                        position: [0, 0],
```
[the remark this answers is not in the document]
Comment (rouzwelt) [reply crates/common/src/lsp_services.rs:59]: well errors that dont have definitive positions as an standard are usually set to the top of the doc, and personally I dont like the big red underline covering all of the text,

**Review by thedavidmeister — APPROVED** [review]

File: crates/common/src/lsp_services.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (mattyg) [reply crates/common/src/lsp_services.rs:59]: I disagree -- it makes its unlikely that users will even notice they have linting errors.

This is how an error in linting the frontmatter appears to users in this PR
image

My preference would be to determine the *actual* position of the error. 

Without that it is preferable to make the error obvious by underlining the entire relevant section i.e. 
image

**Review by mattyg — CHANGES_REQUESTED** [review]
just realized the `parseDotrain` function is no longer used and can be deleted. I still feel like the unknown error locations could be clearer to the user. Otherwise looks good.

File: crates/common/src/lsp_services.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (mattyg) [reply crates/common/src/lsp_services.rs:59]: We'll merge as-is for now and collect user feedback.

**Review by mattyg — APPROVED** [review]

**Pull request merged**